c/zss.c

```diff
@@ -1162,13 +1164,36 @@ static bool readAgentHttpsSettingsV2(ShortLivedHeap *slh,
   TlsSettings *settings = (TlsSettings*)SLHAlloc(slh, sizeof(*settings));
   settings->maxTls = jsonObjectGetString(httpsConfigObject, "maxTls");
   settings->minTls = jsonObjectGetString(httpsConfigObject, "minTls");
-  char *ciphers = jsonObjectGetString(httpsConfigObject, "ciphers");
+  
+  Json *cipherJson = jsonObjectGetPropertyValue(httpsConfigObject, "ciphers");
+  char *ciphers = NULL;
+  if (jsonIsString(cipherJson)) {
   /* 
-   * Takes a string of ciphers. This isn't ideal, but any other methods are
-   * going to be fairly complicated.
-   *
+   * Takes a string of ciphers.
    * ciphers: 13021303003500380039002F00320033
    */
+    ciphers = jsonObjectGetString(httpsConfigObject, "ciphers");
+    zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG, "Cipher string override to %s\n", ciphers);
+  } else {
+    JsonArray *cipherArray = jsonObjectGetArray(httpsConfigObject, "ciphers");
+    int count = jsonArrayGetCount(cipherArray);
+    ciphers = (char *)safeMalloc((4*count)+1, "cipher list");
+    for (int i = 0; i < count; i++) {
+      char *ianaName = jsonArrayGetString(cipherArray, i);
+      zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG, "Cipher request=%s\n", ianaName);
+      CipherMap *cipher = (CipherMap *)ianaCipherMap;
+      while (cipher->suiteId != NULL) {
```

Comment:
This is a nitpick, but it might be better to calculate the total number of elements and use that for a loop as opposed to relying on your static array ending with a null.

Reply:
that makes sense